### PR TITLE
Bugfix/fix 453

### LINKED ISSUE #453: [BUG] using .head(2) to validate scorers fails when sample contains single label
This happens inside `validate_fitting` when should_return_array = False.

Running `self._run_score(model, df.head(2), dataset)` when head contains only single kind of label raises the following bug:

> TypeError: issubclass() arg 2 must be a class or tuple of classes

### PR DESCRIPTION
**Pull request by noamzbr**
resolves #453

### CODE REVIEW

File: deepchecks/utils/metrics.py [hunk 1]
```diff
@@ -157,7 +157,9 @@ def validate_fitting(self, model, dataset: 'base.Dataset', should_return_array:
                                                   f'returned only {len(result)} elements in the score array value')
 
         else:
-            result = self._run_score(model, df.head(2), dataset)
+            # In order for scorer to return array in right length need to pass him samples from all labels
+            single_label_data = df.groupby(dataset.label_name).head(1)
```
Comment (matanper) [inline deepchecks/utils/metrics.py:161]: in here the scorer should return number so no need for the group by (and also comment is incorrect)

File: tests/checks/performance/performance_report_test.py [hunk 2]
```diff
@@ -69,6 +70,26 @@ def test_classification(iris_split_dataset_and_model):
                 metric_col = class_col.loc[class_col['Metric'] == metric]
                 assert_that(metric_col['Value'] , close_to(1, 0.3))
 
+
+def test_classification_binary(iris_dataset_single_class_labeled):
+    # Arrange
+    train, test = train_test_split(iris_dataset_single_class_labeled.data, test_size=0.33, random_state=42)
+    train_ds = iris_dataset_single_class_labeled.copy(train)
```
Comment (matanper) [inline tests/checks/performance/performance_report_test.py:77]: why do you have to copy?

File: tests/checks/performance/performance_report_test.py [hunk 3]
```diff
@@ -69,6 +70,26 @@ def test_classification(iris_split_dataset_and_model):
...
+    clf = RandomForestClassifier(random_state=0)
+    clf.fit(train_ds.features_columns, train_ds.label_col)
+    check = PerformanceReport()
+
+    # Act X
+    result = check.run(train_ds, test_ds, clf).value
+    # Assert
+    for dataset in ['Test', 'Train']:
+        dataset_col = result.loc[result['Dataset'] == dataset]
+        for metric in ['Accuracy (Default)', 'Precision (Default)', 'Recall (Default)']:
```
Comment (matanper) [inline tests/checks/performance/performance_report_test.py:88]: maybe use the default dict `keys()` for future changes

File: deepchecks/utils/metrics.py (hunk 1 above)
Comment (noamzbr) [reply deepchecks/utils/metrics.py:161]: The groupby fixes the bug - in order to run, the sklearn scorer must have at least one sample of each label

About the comment-  will fix

File: tests/checks/performance/performance_report_test.py (hunk 2 above)
Comment (noamzbr) [reply tests/checks/performance/performance_report_test.py:77]: It's a way to split to train and test a dataset that arrived un-splitted

File: tests/checks/performance/performance_report_test.py (hunk 3 above)
Comment (noamzbr) [reply tests/checks/performance/performance_report_test.py:88]: fixed

File: tests/checks/performance/performance_report_test.py (hunk 2 above)
Comment (matanper) [reply tests/checks/performance/performance_report_test.py:77]: how is that a split?

Comment (noamzbr) [reply tests/checks/performance/performance_report_test.py:77]: first I do train_test_split, then I create a dataset from each of the splitted dataframes copying the metadata from the original dataset

**Review by matanper — APPROVED** [review]

**Pull request merged**